flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/PatternStreamBuilder.java

```diff
@@ -60,14 +59,28 @@
 	 */
 	private final OutputTag<IN> lateDataOutputTag;
 
+	private boolean isProcessingTime;
```

Comment:
I think it would be better to introduce an enum for this. Sth like `TimeCharacteristic` that has `PROCESSING_TIME` and `EVENT_TIME`. I find it's often hard to see what `true` or `false` mean when you just read a method call.

This should be internal, though, and the only API for users should be `inEventTime()` and `inProcessingTime()`.

Reply:
@aljoscha I previously consider the enum like `TimeCharacteristic`, but this enum is only used in this which makes little sense and confused for users in 1.11 version.

Reply:
Users wouldn't see it because it's only internal. And having an enum will be more maintainable because, again, just seeing `true` or `false` in code makes it very hard to know what it means.

Reply:
@aljoscha Yeah. I would like to add enum for processing time and event time.